src/control/tools/EditSelection.cpp

```diff
@@ -48,7 +50,8 @@ EditSelection::EditSelection(UndoRedoHandler* undo, Selection* selection, XojPag
     view->rerenderPage();
 }
 
-EditSelection::EditSelection(UndoRedoHandler* undo, Element* e, XojPageView* view, const PageRef& page) {
+EditSelection::EditSelection(UndoRedoHandler* undo, Element* e, XojPageView* view, const PageRef& page):
+        SnapToGridInputHandler(view->getXournal()->getControl()->getSettings()) {
```

Comment:
Use composition over inheritance here, since this is a "has-an" not an "is-an" relationship.